Code/src/main/java/nl/utwente/viskell/ui/components/ConnectionAnchor.java

```diff
@@ -161,7 +80,46 @@ public final String getStringType() {
     protected void handleConnectionChanges(boolean finalPhase) {
         this.block.handleConnectionChanges(finalPhase);
     }
+
+    private void handleMousePress(MouseEvent event) {
+    	if (this.wireInProgess == null && !event.isSynthesized()) {
+    		this.wireInProgess = DrawWire.initiate(this.block.getPane(), this, DrawWire.INPUT_ID_MOUSE);
+    		event.consume();
+    	}
+    }
+
+    private void handleTouchPress(TouchEvent event) {
+    	if (this.wireInProgess == null) {
+    		int touchID = event.getTouchPoint().getId();
+    		this.wireInProgess = DrawWire.initiate(this.block.getPane(), this, touchID);
+    		event.consume();
+    	}
+    }
     
+    private void handleMouseDrag(MouseEvent event) {
+    	if (this.wireInProgess != null && !event.isSynthesized()) {
+    		this.wireInProgess.handleMouseDrag(event);
+    	}
+    }
+    
+    private void handleMouseRelease(MouseEvent event) {
+    	if (this.wireInProgess != null && !event.isSynthesized()) {
+    		this.wireInProgess.handleMouseRelease(event);
+    	}
+    }
+
+    private void handleTouchMove(TouchEvent event) {
+    	if (this.wireInProgess != null) {
+    		this.wireInProgess.handleTouchMove(event);
+    	}
+    }
+    
+    private void handleTouchRelease(TouchEvent event) {
+    	if (this.wireInProgess != null) {
+    		this.wireInProgess.handleTouchRelease(event);
+    	}
+    }
+
```

Comment:
These `null` checks could be `ifPresent` if `wireInProgress` were an `Optional`.